server/routes/offenceCodeDecisions/offenceCodeDecisions.ts

```diff
@@ -80,4 +98,57 @@ export default class OffenceCodeRoutes {
       })
     )
   }
+
+  private async placeholderValues(adjudicationNumber: string, user: User) {
+    const draftAdjudication = await this.placeOnReportService.getDraftAdjudicationDetails(
+      Number(adjudicationNumber),
+      user
+    )
+    const [prisonerDetails, associatedPrisoner] = await Promise.all([
+      this.prisonerDetails(draftAdjudication.draftAdjudication.prisonerNumber, user),
+      this.prisonerDetails(draftAdjudication.draftAdjudication?.incidentRole?.associatedPrisonersNumber, user),
+    ])
+    return {
+      offenderFirstName: properCaseName(prisonerDetails?.firstName),
+      offenderLastName: properCaseName(prisonerDetails?.lastName),
+      assistedFirstName: properCaseName(associatedPrisoner?.firstName),
+      assistedLastName: properCaseName(associatedPrisoner?.lastName),
+    }
+  }
+
+  // If a member of staff or prisoner has been searched for then additional view data is displayed.
+  private async viewDataFromDecisionForm(form: DecisionForm, user: User) {
+    const userId = form?.selectedDecisionData?.userId // This could be a staff username or a prisoner number
+    if (form?.selectedDecisionData?.userId) {
+      switch (decisionTree.findById(form.selectedDecisionId).getType()) {
+        case DecisionType.OFFICER:
+        case DecisionType.STAFF: {
+          const decisionStaff = await this.userService.getStaffFromUsername(userId, user)
+          return { staffName: properCaseName(decisionStaff.name) }
+        }
+        case DecisionType.PRISONER: {
+          const decisionPrisoner = await this.prisonerDetails(userId, user)
+          return { prisonerName: formatName(decisionPrisoner.firstName, decisionPrisoner.lastName) }
+        }
+        default:
+          break
+      }
+    }
+    return {}
+  }
+
+  private async prisonerDetails(prisonerNumber: string, user: User) {
+    if (prisonerNumber) {
+      return this.placeOnReportService.getPrisonerDetails(prisonerNumber, user)
+    }
+    return null
+  }
+
+  private redirect(pathAndQuery: { pathname: string; query: { [key: string]: string } }, res: Response) {
+    return res.redirect(url.format(pathAndQuery))
+  }
+
+  private urlHere(req: Request) {
+    return `/offence-code-selection${req.path}`
```

Comment:
Should we constantize things like 'offence-code-selection' as they are used in a few files in the code base?